extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcConfig.java

```diff
@@ -37,6 +38,12 @@
     @ConfigItem
     public TokenCache tokenCache = new TokenCache();
 
+    /**
+     * Config for well known OIDC providers
+     */
+    @ConfigItem
+    public ProvidersConfig provider;
```

Comment:
I'm not sure it has to be decoupled from `OidcTenantConfig`, it has to be part of `OidcTenantConfig`, tenant specific as needed, or rather, IMHO, something like `quarkus.oidc.provider=github` should point to an instance of `OidcTenantConfig`, merged with the `OidcTenantConfig` referencing it, please see below

Reply:
Also, just realized, this way won't work with the dynamic `TenantConfigResolver` - one should be able to do, instead of defining this config statically in `application.properties`,
```
public class MyTenantConfigResolver implements TenantConfigResolver {
    public Uni<OidcTenantConfig> resolve(RoutingContext rc) 
          {
             OidcTenantConfig github = getDefaultOidcTenantGithubBlock();
             github.getAuthentication().setScope("user:something");
             return github;
          }
    }
}
```

Reply:
@stuartwdouglas Stuart, how about having an `OidcTenantConfig` `provider` enum property with (`github`, etc, values).

This will allow `OidcRecorder`, when going over every `OidcTenantConfig`, pick up a referenced well-known provider `OidcTenantConfig` instance and merge with the current `OidcTenantConfig` - it will address all the concerns I've expressed in various comments here